bonus/graphql-loader/README.md

```diff
@@ -0,0 +1,278 @@
+  developers {
+    name
+    builtProjects {
+      name
+    }
+  }
+}
+```
+
+So far, so good! You should see the server's output on the right side of the GraphiQL interface.
+However, if we look at the server's logs, we can see that the server has executed 4 SQL queries:
+
+```json
+{"level":40,"hostname":"Eomm","msg":"sql: SELECT * FROM Developers"}
+{"level":40,"hostname":"Eomm","msg":"sql: SELECT * FROM Projects WHERE devId = 1"}
+{"level":40,"hostname":"Eomm","msg":"sql: SELECT * FROM Projects WHERE devId = 2"}
+{"level":40,"hostname":"Eomm","msg":"sql: SELECT * FROM Projects WHERE devId = 3"}
+```
+
+As you can see, the queries are not optimized because we ran a query to fetch the projects for each developer instead of fetching all the projects in a single query.
+
+Now you have seen the N+1 problem in action:
+
+- **1**: we run a root query to fetch the first data list
+- **+N**: we run a query for each item of the previous list to fetch the related data
+
+So, if we had 100 developers, we would run 101 queries instead of 2!
+Now that we have seen the problem, let's solve it.
+
+### How to solve the N+1 problem?
+
+The most common way to solve the N+1 problem is to use **DataLoaders**.
+The DataLoader allows you to batch and cache the results of your queries and reuse them when necessary.
+
+Mercurius offers you two ways to use DataLoader:
+
+- [**Loader**](https://github.com/mercurius-js/mercurius/blob/HEAD/docs/loaders.md): it is a built-in DataLoader-Like solution that is quick to setup and use.
+- [**DataLoader**](https://github.com/graphql/dataloader): it is the standard solution to N+1 problem.
+
+In this article, we are going to see both solutions and compare them.
+
+#### Mercurius Loader in action
+
+The `loader` feature is a built-in DataLoader-Like solution that is quick to set up and use.
+It **replaces** Mercurius' `resolvers` option.
+
+Let's see how to use it by optimizing the previous `app.js` example:
+
+```js
+// ... previous code
+
+const loaders = {
+  Developer: {
+    builtProjects: async function loader(queries, context) {
+      const devIds = queries.map(({ obj }) => obj.id)
+      const sql = `SELECT * FROM Projects WHERE devId IN (${devIds.join(',')})`
+      context.app.log.warn('sql: %s', sql)
+
+      const projects = await context.app.sqlite.all(sql)
+      return queries.map(({ obj }) => {
+        return projects.filter((p) => p.devId === obj.id)
+      })
+    },
+  },
+}
+
+const resolvers = {
+  Query: {
+    // ... previous code
+  },
+  Developer: {
+    // ❌ Delete the `builtProjects` resolver. It would be ignored in any case
+  },
+}
+
+// ... previous code
+
+app.register(mercurius, {
+  schema: gqlSchema,
+  graphiql: true,
+  loaders, // add the loaders option
+  resolvers,
+})
+```
+
+As you can see, we have replaced the `resolvers.Developer.builtProjects` function with the `loaders` one.
+The difference is that the `loaders` receives an array of queries instead of a single `parent` object.
```

Comment:
```suggestion
The difference is that the `loaders` receives an array of queries (results from the parent query) instead of a single `parent` object.
```